project/CONTRIBUTING.md

```diff
@@ -186,6 +186,25 @@ interface BarInterface
 }
 ```
 
+In some cases, you will have the possibility to warn the user that things will change,
+and recommend a new way of doing things. You can do so by triggering the dedicated kind of error, like this:
+
+```php
+<?php
+if (/* some condition showing the user is using the legacy way */) {
+    @trigger_error(
+        'Some message about this usage being deprecated in favor of something else'.
```

Comment:
>  links to good examples present on our code

Nah, don't make people click : it's far and you risk a 404. I'll give a proper example instead.